### DISCUSSION TITLE
[Request] Blocky

### DISCUSSION THREAD

**Discussion by DSDV** [tteck/Proxmox#309] [Ideas]
Hi, 

id like to request a new script for Blocky.
https://github.com/0xERR0R/blocky
It is a little like PiHole but takes far less resources and has "better"/other features as well
The documentation can be found here:
https://0xerr0r.github.io/blocky/installation/

big thanks <3

**Comment by DSDV** [comment]
@tteck any chance to donate? <3
i have PayPal (:

**Reply by DSDV** [reply]
@tteck any news on this? (Blocky & PayPal?:)

**Reply by nacho-domatix** [reply]
> @tteck any chance to donate? <3 i have PayPal (:

<a href='https://ko-fi.com/D1D7EP4GF'><img src='https://img.shields.io/badge/☕-Buy me a coffee-red' /></a>   

If you tried any LXC container, it's in there. 

I don't think this software can be a priority, as there are already two adblockers (Pihole and AdGuard) and Blocky can be installed using Docker (there is a docker LXC too).

Anyway, that's my opinion, I'm not involved in this project at all.

**Reply by DSDV** [reply]
sadly adguard is a ad business promoting subscriptions and blocky is super slim and performant while additionally offering reverse dns services

**Reply by nacho-domatix** [reply]
That might be right, but AdGuard is a free and open source project, and it works! If you don't like it (I don't), you can always have PiHole which is an extraordinary community driven software, and Technitium (I did't try that one).

I don't see the necessity of adding a new software that pretty much does the same as other 3 ones already working. I'd prefer go for more alternatives to Photoprism, or new software that provide functionalities that aren't covered yet by the scripts.

Disclaimer: again, I'm not developer nor involved in this project at all.

**Reply by DSDV** [reply]
Technitium is a .NET tool. Since (especially at time of high resource cots) i strive towards performance worth per cots solutions pihole might be a cool foss project but Blocky is a better performer. which is there for the best cost alternative and a good value add to the scripts. 

if you are rich and the finite resources on our planet don't matter to you that is of course another story ;)

**Reply by nacho-domatix** [reply]
Agree, with infinite resources create and maintain 4 differents scripts for the same purpose might be great. But is not the situation, IMO efforts should focus on give new value to this project as we already have 3 different software that are doing what you demand.

Pihole has almost 40k stars in github, AdGuard 13k, Blocky does not even reach 2k. Consider that your opinion might be not shared with majority, and of course you're free to use it, but I understand time is precious to the team and they might be focus in add new value to this project by creating and maintaining scripts.

Also for sure there are more adblock "apps" that for some people are better than Blocky, and they might be right, but the point (again IMHO) should be find new functionalities and even alternatives for software which doesn't have one.

**Reply by DSDV** [reply]
so what you are saying is only because s.th. is a minority its bad? Like Linux vs Microsoft? 
Or god forbid other examples in history or mankind i wont name?... im not following that. 

History shows that more often than not the better solutions are not the most prominent.

Also Blocky adds better and new (additional) features ;)

**Reply by nacho-domatix** [reply]
When did I say is bad?

Use it if you love it, please. But consider most of Linux, open source and Proxmox users might not like it or prefer another options. That's it.

**Comment by tteck** [comment]
@DSDV I've got a test script for you to try
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/blocky-v4.sh)"
```
after install, run `nano /opt/blocky/config.yml` to make any changes
to start Blocky `cd /opt/blocky && ./blocky --config config.yml`

Haven't decided if it will make the main branch yet.

**Reply by DSDV** [reply]
wow nice <3
thank you so much - ill try it as soon as possible. 
if you dm me your papal i ll donate as promised (:

**Reply by tteck** [reply]
I'd fork this repo if I was you. If it doesn't make the main branch it could/will get deleted. I personally dislike non-GUI apps, not to mention the headaches I see with others using such an advanced piece of software. 😜

**Reply by DSDV** [reply]
oh ok good to know (: 
there will be frontends in the future. As for know there is a Prometheus and Grafana integration (:

**Reply by tteck** [reply]
Went ahead and created a `blocky.service` file so It'd start at boot. So now stop blocky (`systemctl stop blocky`), make changes to `config.yml` and start (`systemctl start blocky`) it. You can check the status `systemctl status blocky`

**Reply by tteck** [reply]
I went ahead and made it main worthy. 😵‍💫

**Comment by devzwf** [comment]
there is a front-end but very minimal.....if not useless.... but hey it surely will evolved...
and i am all agree with tteck about the headaches with others using such an advanced piece of software

**Comment by SimonMcN** [comment]
I may be delusional but does this work please ?
I have found that port 53 is blocked by systemd-resolved  I have to do service systemd-resolved stop before it will be able to accept queries on port 53 ?

Cheers

**Comment by SimonMcN** [comment]
sorry, should have said better solution is
systemctl stop systemd-resolved
systemctl disable systemd-resolved.service
